plugins/modules/ndo_fabric_span_session_source.py

```diff
@@ -0,0 +1,741 @@
+    description:
+    - The SPAN Drop Packets of the SPAN Session source.
+    - Defaults to O(span_drop_packets=false) when unset during creation.
+    - The O(filter_epg) and O(filter_l3out) is not configurable when this parameter set to true.
+    type: bool
+  filter_epg:
+    description:
+    - The Filter EPG of the SPAN Session source.
+    - When the Filter EPG is specified in the configuration, the Filter L3Out will be removed.
+    - Filter L3Out and Filter EPG cannot be configured simultaneously.
+    type: dict
+    suboptions:
+      enabled:
+        description:
+        - This parameter is used to clear the Filter EPG configuration.
+        type: bool
+      epg_uuid:
+        description:
+        - The UUID of the EPG used to configure the Filter EPG.
+        - This parameter or O(filter_epg.epg) is required.
+        type: str
+      epg:
+        description:
+        - The EPG object detail used to configure the Filter EPG.
+        - This parameter or O(filter_epg.epg_uuid) is required.
+        type: dict
+        suboptions:
+          name:
+            description:
+            - The name of the EPG.
+            type: str
+            required: true
+          template:
+            description:
+            - The name of the template that contains the EPG.
+            - This parameter or O(filter_epg.epg.template_id) is required.
+            type: str
+          template_id:
+            description:
+            - The ID of the template that contains the EPG.
+            - This parameter or O(filter_epg.epg.template) is required.
+            type: str
+          schema_id:
+            description:
+            - The ID of the schema that contains the EPG.
+            - This parameter or O(filter_epg.epg.schema) is required.
+            type: str
+          schema:
+            description:
+            - The name of the schema that contains the EPG.
+            - This parameter or O(filter_epg.epg.schema_id) is required.
+            type: str
+          anp:
+            description:
+            - The name of the ANP that contains the EPG.
+            - This parameter or O(filter_epg.epg.anp_uuid) is required.
+            type: str
+          anp_uuid:
+            description:
+            - The UUID of the ANP that contains the EPG.
+            - This parameter or O(filter_epg.epg.anp) is required.
+            type: str
+  filter_l3out:
+    description:
+    - The Filter L3Out of the SPAN Session source.
+    - When the Filter L3Out is specified in the configuration, the Filter EPG will be removed.
+    - Filter L3Out and Filter EPG cannot be configured simultaneously.
+    type: dict
+    suboptions:
+      enabled:
+        description:
+        - This parameter is used to clear the Filter L3Out configuration.
+        type: bool
+      tenant:
+        description:
+        - The name of the tenant. This parameter is used to associate the L3Out from APIC.
+        type: str
+      l3out:
```

Comment:
Can this be a dictionary as discussed in the other comments. It would be consistent with the `epg` option.